packages/teams-js/test/public/settings.spec.ts

```diff
@@ -138,120 +209,154 @@ describe('settings', () => {
-    expect(message).not.toBeNull();
-    expect(message.args.length).toBe(0);
-  });
+      let handlerCalled = false;
+      settings.registerOnSaveHandler(saveEvent => {
+        saveEvent.notifyFailure('someReason');
+        handlerCalled = true;
+      });
 
-  it('should successfully notify failure from the registered save handler', async () => {
-    await utils.initializeWithContext('settings');
+      utils.sendMessage('settings.save');
 
-    let handlerCalled = false;
-    settings.registerOnSaveHandler(saveEvent => {
-      saveEvent.notifyFailure('someReason');
-      handlerCalled = true;
+      expect(handlerCalled).toBe(true);
+      const message = utils.findMessageByFunc('settings.save.failure');
+      expect(message).not.toBeNull();
+      expect(message.args.length).toBe(1);
+      expect(message.args[0]).toBe('someReason');
     });
 
-    utils.sendMessage('settings.save');
+    it('settings.registerOnSaveHandler should not allow multiple notifies from the registered save handler', async () => {
+      await utils.initializeWithContext(FrameContexts.settings);
 
-    expect(handlerCalled).toBe(true);
-    const message = utils.findMessageByFunc('settings.save.failure');
-    expect(message).not.toBeNull();
-    expect(message.args.length).toBe(1);
-    expect(message.args[0]).toBe('someReason');
+      let handlerCalled = false;
+      settings.registerOnSaveHandler(saveEvent => {
+        saveEvent.notifySuccess();
+        expect(() => saveEvent.notifySuccess()).toThrowError('The SaveEvent may only notify success or failure once.');
+        expect(() => saveEvent.notifyFailure()).toThrowError('The SaveEvent may only notify success or failure once.');
+        handlerCalled = true;
+      });
+
+      utils.sendMessage('settings.save');
+
+      expect(handlerCalled).toBe(true);
+      const message = utils.findMessageByFunc('settings.save.success');
+      expect(message).not.toBeNull();
+      expect(message.args.length).toBe(0);
+    });
   });
 
-  it('should successfully notify success on remove when there is no registered handler', async () => {
-    await utils.initializeWithContext('remove');
+  describe('Testing settings.registerOnRemoveHandler function', () => {
+    const allowedContexts = [FrameContexts.remove, FrameContexts.settings];
+    Object.values(FrameContexts).forEach(context => {
+      if (!allowedContexts.some(allowedContexts => allowedContexts == context)) {
+        it(`settings.registerOnRemoveHandler does not allow calls from ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          expect(() => settings.registerOnRemoveHandler(() => true)).toThrowError(
+            `This call is only allowed in following contexts: ${JSON.stringify(
+              allowedContexts,
+            )}. Current context: "${context}".`,
+          );
+        });
+      }
+    });
+
+    it('should successfully register a remove handler', async () => {
```

Comment:
I don't think this test ('should successfully register a remove handler') is actually verifying anything?